service/frontend/workflow_handler.go

```diff
@@ -4186,28 +4186,72 @@ func (wh *WorkflowHandler) getHistory(
 	branchToken []byte,
 ) (*historypb.History, []byte, error) {
 
+	// readFullPageEvents reads a full page of history events from history service. Due to storage format of V2 History
+	// it is not guaranteed that pageSize amount of data is returned. Function returns the list of history events, the
+	// size of data read, the next page token, and an error if present.
+	readFullPageEvents := func(
```

Comment:
nit: Extract fn instead of using lambda

Reply:
this is what I mentioned a while back... recently the linter now prohibits functions with more than 3 return values. Given this logic is a direct port from `persistence.ReadFullPageEvents()`. I think it's better to keep those two implementations untouched and a mirror of each other.